lib/primer/classify/attr_cache.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+require_relative "flex"
+
+module Primer
+  class Classify
+    # :nodoc:
+    class AttrCache
+      include Singleton
+
+      def initialize
+        @cache_enabled = true
+        @lookup = {}
+      end
+
+      private :initialize
+
+      def fetch(breakpoint, key, val)
+        found = @lookup.dig(breakpoint, key, val)
+        return found if found
+
+        yield.tap do |result|
+          set(result, breakpoint, key, val) if @cache_enabled
+        end
+      end
+
+      def disable
```

Comment:
Less of an immediate issue and more of something to be aware of, but this isn't thread-safe. Should we document that somewhere, maybe as part of some basic module docs?

Reply:
Hmm... we could, although it's supposed to only be used in tests.